pkg/configuration/configuration.go

```diff
@@ -115,32 +133,73 @@ func CreateConfigurationFile(filename string) (string, error) {
 	return p, err
 }
 
+type Opts = func(config Configuration)
+
+func WithAutomaticEnv() Opts {
+	return func(c Configuration) {
+		c.AutomaticEnv()
+	}
+}
+
+func WithSupportedEnvVars(envVars ...string) Opts {
+	return func(c Configuration) {
+		c.SetSupportedEnvVars(envVars...)
+	}
+}
+
+func WithSupportedEnvVarPrefixes(prefixes ...string) Opts {
+	return func(c Configuration) {
+		c.SetSupportedEnvVarPrefixes(prefixes...)
+	}
+}
+
+func WithFiles(files ...string) Opts {
+	return func(c Configuration) {
+		c.SetFiles(files...)
+	}
+}
+
+// NewWithOpts creates a new snyk configuration file with optional parameters
+func NewWithOpts(opts ...Opts) Configuration {
+	config := createViperDefaultConfig(opts...)
+	return config
+}
+
 // New creates a new snyk configuration file.
 func New() Configuration {
-	config := NewFromFiles("snyk")
+	config := NewWithOpts(
+		WithFiles("snyk"),
+		WithAutomaticEnv(),
+	)
 	return config
 }
 
+// Deprecated: Use NewWithOpts with configuration.WithFiles() and configuration.WithAutomaticEnv() options instead
+//
 // NewFromFiles creates a new Configuration instance from the given files.
 func NewFromFiles(files ...string) Configuration {
-	config := createViperDefaultConfig()
-	config.configType = jsonFile
-	readConfigFilesIntoViper(files, config)
+	config := NewWithOpts(
+		WithFiles(files...),
+		WithAutomaticEnv(),
+	)
 	return config
 }
 
 func (ev *extendedViper) getConfigType() configType {
 	return ev.configType
 }
 
+// Deprecated: Use NewWithOpts with configuration.WithAutomaticEnv() option instead
```

Comment:
Question: I'm still a bit undecided myself, but let me still raise this question, do you think it is essential to deprecate these helpers? I'm asking since I understand the wish to keep the interface clean but on the other hand it creates quite some noise here and in the other PR. Wouldn't it be better to have the tests for example still use the old methods and ensure that the changes don't cause a change in behaviour.
From a reviewer perspective it would make it easier for sure.

Reply:
It's not essential, but IMO it's better to identify and ideally address these issues early, rather than defer them. Marking as deprecated seems a good compromise between identifying and fixing now vs keeping the status quo; it gives us the opportunity to point people to the desired implementation and hopefully avoids building tech debt.

Ideally, marking as deprecated would've been the only change, but the linter interpreted it as a fail, perhaps it would've been better to see if it was possible to change the lint error to a warn.